### PR TITLE
OTel Lambda go instrumentation docs

### PR DESCRIPTION
**Pull request by bhautikpip**
Draft as awaiting merge of instrumentation SDK to OpenTelemetry Go Contributions Repo

### CODE REVIEW

**Comment by garrettwegan** [comment]
Blocked on [contrib #983](https://github.com/open-telemetry/opentelemetry-go-contrib/pull/983) and [contrib #984](https://github.com/open-telemetry/opentelemetry-go-contrib/pull/983)

TODO: 
* Check the various links are valid once contrib PRs are merged (especially pkg.go.dev links, assume they are automatic but unsure)

**Comment by alolita** [comment]
@willarmiros what is the status on this PR? are the PRs merged upstream?

**Comment by alolita** [comment]
Blocked on upstream merge. Will look at later when upstream work is ready.

**Review by alolita — COMMENTED** [review]
Diagram needs to be fixed. Not viewable.

**Comment by bhautikpip** [comment]
> Diagram needs to be fixed. Not viewable.

How can I validate that? Looks like we use the same image as [.NET lambda](https://github.com/aws-otel/aws-otel.github.io/search?q=img16) and .NET lambda image can be seen [here](https://aws-otel.github.io/docs/getting-started/lambda/lambda-dotnet)

File: src/docs/getting-started/lambda/lambda-go.mdx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+```shell
+aws lambda update-function-configuration --function-name <FUNCTION> --runtime provided.al2
+```
+
+Re-upload source zip with function executable renamed to `bootstrap`
+
+
+## Instrumentation
+
+### Code Instrumentation
```
Comment (willarmiros) [inline src/docs/getting-started/lambda/lambda-go.mdx:37]: We need to add a new step here to add the boilerplate code to a Lambda function. It should be step 1 to add the imports (including new ones as needed for the extra step), step 2 to copy in the code for creating and starting a TracerProvider, then step 3 is instrumenting the handler.

Comment (willarmiros) [reply src/docs/getting-started/lambda/lambda-go.mdx:37]: We should also probably add a step 4 for instrumenting downstream requests too - you can just link to https://aws-otel.github.io/docs/getting-started/go-sdk/trace-manual-instr#instrumenting-an-application

File: src/docs/getting-started/lambda/lambda-go.mdx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  which runs as a Lambda extension.
+
+Note: Lambda layers are a rationalized resource, meaning that they can only be used in the Region in which they are published.
+ Make sure to use the layer in the same region as your Lambda functions.
+
+Find the supported regions and ARN in the table below for the ARNs to consume.
+
+|Supported   Regions |Lambda layer ARN format| Contents |
+|---------------------|-------------------------|----------|
+| ap-northeast-1<br/>ap-northeast-2<br/>ap-south-1<br/>ap-southeast-1<br/>ap-southeast-2<br/>ca-central-1<br/>eu-central-1<br/>eu-north-1<br/>eu-west-1<br/>eu-west-2<br/>eu-west-3<br/>sa-east-1<br/>us-east-1<br/>us-east-2<br/>us-west-1<br/>us-west-2 | arn:aws:lambda:\<region\>:901920570463:layer:aws-otel-collector-ver-0-36-0:1 | Contains the [ADOT Collector for Lambda v0.11.0](https://github.com/aws-observability/aws-otel-collector/releases/tag/pkg%2Flambdacomponents%2Fv0.11.0)|
```
Comment (willarmiros) [inline src/docs/getting-started/lambda/lambda-go.mdx:66]:
```suggestion
| ap-northeast-1<br/>ap-northeast-2<br/>ap-south-1<br/>ap-southeast-1<br/>ap-southeast-2<br/>ca-central-1<br/>eu-central-1<br/>eu-north-1<br/>eu-west-1<br/>eu-west-2<br/>eu-west-3<br/>sa-east-1<br/>us-east-1<br/>us-east-2<br/>us-west-1<br/>us-west-2 | arn:aws:lambda:\<region\>:901920570463:layer:aws-otel-collector-ver-0-39-0:1 | Contains the [ADOT Collector for Lambda v0.15.0](https://github.com/aws-observability/aws-otel-collector/releases/tag/pkg%2Flambdacomponents%2Fv0.15.0)|
```

**Review by willarmiros — CHANGES_REQUESTED** [review]
Going to need more code for the customer to copy into their `main` function, like we have in our sample app, since it's no longer a 1-line experience.

File: src/docs/getting-started/lambda/lambda-go.mdx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+1. Add dependencies for the [ADOT Lambda Go SDK](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-lambda-go/otellambda) and the [recommended SDK configuration options for AWS X-Ray](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-lambda-go/otellambda/xrayconfig).
+
+```go
+import (
+    "context"
+    "github.com/aws/aws-lambda-go/lambda"
+	"go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-lambda-go/otellambda"
+	"go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-lambda-go/otellambda/xrayconfig"
+	"go.opentelemetry.io/contrib/propagators/aws/xray"
+    "go.opentelemetry.io/otel"
```
Comment (willarmiros) [inline src/docs/getting-started/lambda/lambda-go.mdx:48]:
```suggestion
    "github.com/aws/aws-lambda-go/lambda"
    "go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-lambda-go/otellambda"
    "go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-lambda-go/otellambda/xrayconfig"
    "go.opentelemetry.io/contrib/propagators/aws/xray"
    "go.opentelemetry.io/otel"
```

File: src/docs/getting-started/lambda/lambda-go.mdx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+   		}
+   	}(ctx)
+
+   	otel.SetTracerProvider(tp)
+   	otel.SetTextMapPropagator(xray.Propagator{})
+```
+
+3. Wrap handler in call to `lambda.Start()` or `lambda.StartHandler()` in `main()` function using the recommended X-Ray configuration options.
+```go
+   lambda.Start(otellambda.InstrumentHandler(lambda_handler(ctx), xrayconfig.WithRecommendedOptions(tp)... ))
```
Comment (willarmiros) [inline src/docs/getting-started/lambda/lambda-go.mdx:73]:
```suggestion
lambda.Start(otellambda.InstrumentHandler(lambda_handler(ctx), xrayconfig.WithRecommendedOptions(tp)... ))
```

**Review by willarmiros — APPROVED** [review]
Just a few whitespacing issues, otherwise lgtm!

**Comment by alolita** [comment]
@bhautikpip @willarmiros is this PR ready to be merged?

**Comment by willarmiros** [comment]
@alolita yes!

**Review by NathanielRN — APPROVED** [review]

File: src/docs/getting-started/lambda/lambda-go.mdx [hunk 5]
```diff
@@ -0,0 +1,133 @@
+---
+title: 'AWS Distro for OpenTelemetry Lambda Support For Go'
+description:
+    The [AWS Distro for OpenTelemetry Lambda (ADOT Lambda) SDK for Go](https://github.com/open-telemetry/opentelemetry-go-contrib/tree/main/instrumentation/github.com/aws/aws-lambda-go/otellambda)
+     provides provides extension and tracing APIs you can use to instrument your Lambda function.
```
Comment (NathanielRN) [inline src/docs/getting-started/lambda/lambda-go.mdx:5]:
```suggestion
     provides extension and tracing APIs you can use to instrument your Lambda function.
```

**Review by alolita — APPROVED** [review]

**Pull request merged**